lib/mp4/transmuxer.js

```diff
@@ -1213,6 +1213,7 @@ Transmuxer = function(options) {
       videoTrack.timelineStartInfo.dts = undefined;
       videoTrack.timelineStartInfo.pts = undefined;
       clearDtsInfo(videoTrack);
+      pipeline.captionStream.resetDtsTracking();
```

Comment:
Yea this also gets called when contrib-hls detects a discontinuity, which does not require a seek to happen. You'll still want to reset the caption dts tracking on discontinuities.